apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TxTests.cs

```diff
@@ -0,0 +1,220 @@
+                        new SpannerParameterCollection
+                        {
+                            {"K", _key, SpannerDbType.String},
+                            {"StringValue", Guid.NewGuid().ToString(), SpannerDbType.String}
+                        });
+                    _history.Add(new HistoryEntry
+                    {
+                        Value = insupdate.Parameters[1].Value.ToString(),
+                        Timestamp = DateTime.UtcNow
+                    });
+                    await insupdate.ExecuteNonQueryAsync();
+
+                    await Task.Delay(250);
+
+                    // 2nd update
+                    insupdate.CommandText = "UPDATE TX";
+                    insupdate.Parameters[1].Value = Guid.NewGuid().ToString();
+                    _history.Add(new HistoryEntry
+                    {
+                        Value = insupdate.Parameters[1].Value.ToString(),
+                        Timestamp = DateTime.UtcNow
+                    });
+                    await insupdate.ExecuteNonQueryAsync();
+
+                    await Task.Delay(250);
+
+                    // 3rd update
+                    insupdate.Parameters[1].Value = Guid.NewGuid().ToString();
+                    _history.Add(new HistoryEntry
+                    {
+                        Value = insupdate.Parameters[1].Value.ToString(),
+                        Timestamp = DateTime.UtcNow
+                    });
+                    await insupdate.ExecuteNonQueryAsync();
+                }
+            }
+        }
+
+        [Fact]
+        public async Task TestReadStrong()
+        {
+            await WriteSampleRows();
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                await connection.OpenAsync();
+                using (var tx = await connection.BeginReadOnlyTransactionAsync(TimestampBound.Strong))
+                {
+                    var cmd = connection.CreateSelectCommand("SELECT * FROM TX WHERE K=@k",
+                        new SpannerParameterCollection { { "k", _key, SpannerDbType.String } });
+                    cmd.Transaction = tx;
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            Assert.True(reader.GetFieldValue<string>(1) == _history[2].Value);
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public async Task TestReadMin()
+        {
+            ArgumentException caughtException = null;
+
+            await WriteSampleRows();
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                await connection.OpenAsync();
+                try
```

Comment:
Doesn't have to be in the same test.  But it needs to fail because some readonly modes are designed to be "single use" because the time range will change once they are used.  these modes are only sent along with a read command where they create a short lived transaction that is used one time.
using those options here is therefore an error. 

 I've enabled this single use read scenario as a part of calling ExecuteReader.